### DISCUSSION TITLE
Sonoff S60TPG (UK Version) Successfully Flashed

### DISCUSSION THREAD

**Discussion by MalSmalley** [arendst/Tasmota#22554] [Show and tell]
The Sonoff S60 is a great product.  Looks good, has power monitoring and is cheap but its a sealed unit so not apparently easy to flash. 
A good description and partial teardown by Mat Zolnierczyk [[here]](https://notenoughtech.com/home-automation/sonoff-s60/) gets inside the unit by breaking the glued seal but there are not very accessible access points for flashing.
I totally destroyed one unit (by trying to access points on the processor daughter board) and took this a bit further. 
The underside of the main board has some very useful pads available.
![S60UK Main Board](https://github.com/user-attachments/assets/1b3a18de-d247-4e72-870a-52028dab65a4)
Tx & Rx are obvious.  But the first pin on the connection to the processor board is +5V.  (And it is +5V and not 3.3V as there is a voltage regulator or divider on the processor board).  0V is also at the far end of the connection but this is easier accessed by use of the Neutral pin on the plug itself.

On a new, un broken into, unit I accessed these points by drilling a couple of 3.5mm holes in the base and expanding the access to about 5mm x 10mm and was able to solder connecting wires to these 3 points fairly easily.  I did protect the plastic casing with some foil!

![S60UK Access](https://github.com/user-attachments/assets/747293f9-2d39-4a7f-9467-3e6f1d4c7ef6)
![s60UK Wired](https://github.com/user-attachments/assets/0d6c0950-665a-4c34-833e-16b49360bf14)

I used a usb serial converter from a Pi400.  Note that I powered the S60 from a separate 5V rail but the Rx/Tx from the serial converter were at 3.3V as this is what the esp32 chip requires.  

Flashed using esptool.  Holding the button on the side of the S60 on power up to enter boot mode. This does not have to be held down long.  Flashed with `tasmota32c3.factory.bin`.  
My normal process is (as per the esptool instructions)
```
$esptool.py --port /dev/ttyUSB0 read_mac    (to test connection)
$esptool.py --port /dev/ttyUSB0 read_flash 0x00000 0x10000 s60backup.bin    (in case we need to go back!)
$esptool.py --port /dev/ttyUSB0 erase_flash
$esptool.py --port /dev/ttyUSB0 write_flash -fm out 0x0 tasmota32c3.factory.bin   (where the bin has already been downloaded)
```
Flashing took about 90 seconds and , after a repower, the AP point showed up.

The template I used was the same as from the discussion & progress on the [European version.] (https://github.com/arendst/Tasmota/discussions/21255) - Thank You !! 

`{"NAME":"S60UK","GPIO":[1,1,1,1,224,544,1,3104,1,32,1,0,0,0,0,0,0,0,1,1,1,1],"FLAG":0,"BASE":1}`

![S60UK Flashed](https://github.com/user-attachments/assets/ff5e85ad-bd8d-408f-ae71-a86c74670c80)
And finally a bit of 0.5mm Plasticard was cut to shape and glued onto the base to cover the access hole and tidy up the unit.

![S60UK completed](https://github.com/user-attachments/assets/b3760f61-40c6-421b-b002-d12635bf60c8)

**Comment by sfromis** [comment]
Nice way of doing it.

I spotted one detail:
> `read_flash 0x00000 0x10000 s60backup.bin`

Only backing up 64 KB of flash won't give you the opportunity of going back. With ESP32-C3, I usually backup the full 4 MB of flash, up to `0x00400000`

**Reply by MalSmalley** [reply]
Oooh thanks for that!   I've never 'gone back' yet.  But this has saved some head scratching (& swearing!) if I ever do.

**Comment by JBFUK** [comment]
I brought the same devices and opened one up with the same goal.  I have not been able to flash it.  I just get nothing back from the ESP32.  It might be going into boatloader mode as when simply applying 5v one of the LEDs flashes slownly, when applying power with the button pressed the LED does not come on or flash.  I've now gone so far as to remove the ESP32 daughter board and solder wires on to the test pads to allow me to use the reset pin with a button.


![IMG_8991](https://github.com/user-attachments/assets/f4907b1b-4919-4bc6-8c78-d48b6746e314)
![IMG_8992](https://github.com/user-attachments/assets/55ee0df0-2f69-4064-9e0f-63c945907159)
![IMG_8993](https://github.com/user-attachments/assets/2cb8164e-6ff4-4b8a-8e0b-6907f5c282a3)


I've tried two different USB to serial adapters, both in 3.3v mode.  I've tried with the USB adapter providing the 5v for the chip and with an external battery pack providing 5v.

Not sure what's left to try really.  Perhaps I'm unlucky and have somehow damaged the TX port, or it has a pre-existing fault.

**Reply by MalSmalley** [reply]
That's interesting.  I'd love to see how you get it all back together again when you do flash it!
I found the same- just powering on gave the blue led flashing approx once a second.  But power up with button pressed and no leds illuminate.  This is definitely in boatload mode.

Useful photos of the processor board.  Ive just taken the one I 'nadgered' out to compare and can see where I have damaged it.  It may be easier to use the connector pins rather than the test points. The left most on the processor side is Tx & the 2nd left Rx.  5V, Gnd & Key are quire clear on your photos.

In fact on my trashed board I don't get continuity between the connector pins and the on-board test points.  That could be because of the damage to my board - but worth checking.  The Tx/Rx test points I used on the underside of the main board are definitely connected to the connecting pins 1 & 2.  It may be you get some success using these instead.

I've done flashed two of these now, so its definitely possible.

**Reply by JBFUK** [reply]
Well, no luck with flashing the chip in question.  I checked the resistors on the RX/TX pins, they were fine.  I can only conclude the chip is damaged.

Subsequently I've successfully flashed the other three units - so my serial adapter is fine.  For the first I used your method of drilling holes to provide access to the TX/RX/5v terminals.  This worked 'ok', filling the hole with hot glue afterwards.  For the remaining two I opened the plug up, flashed via the pads and the glued the cases closed again.  I found this much cleaner.  It's really no bother opening up the case - you have to apply pressure using an engineering vice, no too much but enough to just slightly deform it. Release the pressure and rotate 45', apply pressure, repeat until eventually the glue joint shears and the case pops open.  To reassemble the two parts will meet up perfectly so it can be glued back together with no visible evidence of it having been opened.

![image](https://github.com/user-attachments/assets/491615ee-7773-457d-964e-3577e53ade96)

![image](https://github.com/user-attachments/assets/ea31c8b5-fa9b-4a13-93b5-07b331a82b8d)

![image](https://github.com/user-attachments/assets/616b4ee6-20e5-497f-8b00-5cd27e842f77)

![image](https://github.com/user-attachments/assets/4b280019-6cac-4a29-842c-afb56f989922)

I'll probably replace the esp32-c3 chip on the faulty board - a dev board can be had for £1.40 from which I can take the processor and have a bunch of spare parts left over.

Now is it worth the effort?  At the price I brought them on sale - £10 each, it's just about worthwhile if you need some devices in a hurry.  If not, or at a higher price point the it's better to simply buy 'athom' devices pre-flashed with Tasmota32 from aliexpress.

**Comment by vtraveller** [comment]
This is great work by all of you.  What I'd like to do is 3D print a jig and use pogo pins to go in.  I'm struggling to accurately target the points to drill as I'm using the photos to judge.  I suspect V1 will end up hacked up as I work this out.

Dimensionally my estimates are in mm, all using the side for the earth pin for X, and the top of the earth pin for Y:

```
Rx = 5.6,  7.2
Tx = 8.4,  6.2
5V = 10.8, 4.8
```

The pic isn't great as getting the camera aligned makes things look off, but I wanted to show you where I'm measuring from.  The idea is to make the jig fit over the power pins so there's no margin for error.

I estimate the pad areas are approximately 1.6mm is diameter.  I want to go in with a 1mm diameter bit, but unless I pop it open and reglue it, or go for a larger hole on V1 I'm afraid I'll miss.  I want to keep it glued simply due to the forces when you pull the plug out of the socket.

Still - I think the idea is solid, and it would mean the holes don't need filling, and can remain for any boot issues down the line.  If anyone has any clues I could use before attempting this crazy idea, please let me know.

![image](https://github.com/user-attachments/assets/4cff4a82-7d2d-4506-bc81-ffef21a3dbed)

**Reply by JBFUK** [reply]
I had a similar idea but as the plug is so easy to open and re-seal with no sign its ever been modified - I concluded it's not worth the effort.  You'll only need to physically access the hardware once to flash and after that all updates (even between Tamara/esp-home) will be OTA.

**Reply by vtraveller** [reply]
What glue did you use or will it push clip?

**Reply by JBFUK** [reply]
I used a little bit of medium CA (thickish superglue).

**Reply by techiris** [reply]
@vtraveller  - I used @JBFUK's method of slowly applying pressure on the glue join and they pop open quite easily, used a set of shifting pliers to do the work with a cloth as padding while rotating after each "snap" of separation, then glued back as @JBFUK mentioned above.
I really don't think the effort is needed for templates or drilling.
![image](https://github.com/user-attachments/assets/c7389a13-b91f-49ce-a1e8-536388717819)

I have even used this project ( [tasmota-blerry](https://github.com/tony-fav/tasmota-blerry) ) to add BLE gateway features to one of the S60's to receive and forward BLE signals to MQTT broker from other devices such as Govee H5075.

I have 12 flashed and operating with no visible sign of change.

**Reply by vtraveller** [reply]
I'm glad it worked for you. I'm flashing and updating a fair bit as I refine my fw. So I wanted a solution where I could dip in and flash or monitor serial as needed. 

It's also a 1min mod now without any visible intrusion, which I like for the cleanliness of the solution. It was also fun to hack where I don't need to solder wires on the original board.

That said, I have specialist gear. Not everyone will have that. I expect most people will find the vice method easier. So I'm not dissing it, just didn't meet my needs.

**Comment by MalSmalley** [comment]
Hi @vtraveller
That was my first thought too.  I spent days trying to get pins lined up and with good contacts.  
I was trying to build a jig that fitted over the 'rounded triangle' shape of the plug.  Gave up & drilled a bigger hole!  
Attached some photos that might help you in your quest from my dissasembled unit.

![IMG_6920](https://github.com/user-attachments/assets/b00279bb-b3bc-4a3c-a9cf-7ab005aac0e1)
![IMG_6922](https://github.com/user-attachments/assets/82a182aa-3403-4357-846d-bbcf89444003)
![IMG_6921](https://github.com/user-attachments/assets/84548216-5129-4dad-ac40-f3112292e687)

**Reply by vtraveller** [reply]
That's fantastic. Many thanks. 

I'm going to locate based off the plug pins. As these are a standard I feel it's easier as a ref point. Slot over the drill jig, then possibly the programming one.

Originally I was going to make a cutting jig to break into the plug without scratching it.

If I create anything useful I'll post it for the community to use.

**Reply by MalSmalley** [reply]
I think the radial distance from the casing too would be a good triangulation.
Using a vernier I'm measuring 4.7mm for the centre point of the Rx & Tx pads (from outside of casing)
and 8.4mm to the centre of the Vcc.   Also 3.2mm centre to centre between the Rx & Tx pads.

Good luck!

**Reply by vtraveller** [reply]
If you're interested, I'm going to import the pcb+case pic into FreeCAD, scale to your super helpful 0.1mm scale, overlay a design, print it.

Then drill for oil and hope. 🤣

**Reply by vtraveller** [reply]
Everything scaled and aligned: +1 degree rotation needed.  When I started adding reference points I noticed that the pcb has different diameters in different places.  I use a measurement of my own case as a way to cross-reference.

I aligned your measurements with mine - they're close.  Except the Vcc which is so off, I'm wondering who's wrong.

![image](https://github.com/user-attachments/assets/e3bf4f0a-1a5f-451c-9d6b-d2202150e4b9)
![image](https://github.com/user-attachments/assets/268e952a-dc5e-4e75-ba39-b2d651269e61)

**Reply by MalSmalley** [reply]
You are correct.  I am wrong.  I forgot which pin on the processor board it was, didn't think to check, and measured to the inner side.
Should be 5.5mm by my (new) measurement to the correct pin!!

**Reply by vtraveller** [reply]
Awesome. Means I'm zoning in. 😆

**Comment by vtraveller** [comment]
Well I think I have a candidate. @MalSmalley if you want to give it a go feel free.  As you have an open one you can see how off I am.  I did struggle getting enough material around Rx, and had to nudge it slightly off centre by a fraction of a mm.

https://www.printables.com/model/1172012-sonoff-s60-test-jig

![image](https://github.com/user-attachments/assets/ccc69e58-f64e-48ca-9f76-f11cf4932753)

**Reply by vtraveller** [reply]
![image](https://github.com/user-attachments/assets/e41b6173-892c-4053-b022-00ee4f27287b)

Well I hope this works. I'm tempted to go for a slightly larger diameter and use 0.1" pitch pins. 

At the moment the jig is using 0.6mm pogos. The holes are 1.1mm as I want some jiggle room - relatively speaking 😅

**Reply by vtraveller** [reply]
I get 1.8k from the +5V hole to gnd. I'm out slightly by 0.8mm though down.

So today's plan is to move the hole and increase its diameter to 1.6mm and erase the mistake. 

Once I have power I can safely probe for Tx Rx without risk of frying anything. I've been unable to get anything Tx or Rx to Gnd. 

I may switch from 0.7mm pogos to IDC header pins such that the jig solution is accessible to more people.

This is feeling like lock picking. Kinda fun.

**Reply by MalSmalley** [reply]
It's the doing it that's fun.   A sense of 'I won't be beaten'.   
I'm measuring 1.9k Vcc-Neutral pin but my processor board disconnected and probably fried!

I used esptool to flash (from an Rpi400)
Just reading from the flash is a good first check without risking erasing the flash or partially installing.

`esptool.py --port /dev/ttyUSB0 read_mac`

Obviously Serial port address may vary depending on your config.

**Reply by vtraveller** [reply]
Ain't that the truth. 😀

Given what people have fried I do have a plan. I'm actually going to apply 5v current limited (a mA or two) and measure draw. I'll then scale up until I see the boot sequence through the current changes. 

If my mods go south, I'll gradually increase the diameter of the holes using jigs until I can see the contact points. 

I'll 3D print a plug for v1 in white, while knowing my remaining S60s will be aligned correctly. 

Thanks for the measurement. I think I've hit the 5V given that circuit should still be working on your fried one.

**Reply by vtraveller** [reply]
Success. Now on to the Rx Tx pins. 

https://youtu.be/_BHexJm0jKI

**Reply by vtraveller** [reply]
Boot strap caught:
![image](https://github.com/user-attachments/assets/d4128372-3e03-4775-8617-78f3ee6135ec)

![image](https://github.com/user-attachments/assets/a853c03f-0712-4610-883a-a269a1be9687)

**Reply by vtraveller** [reply]
Looks like I've managed the hattrick:

```
python -m esptool read_mac
esptool.py v4.8.1
Found 1 serial ports
Serial port COM10
Connecting....
Detecting chip type... ESP32-C3
Chip is ESP32-C3 (QFN32) (revision v0.4)
Features: WiFi, BLE, Embedded Flash 4MB (XMC)
Crystal is 40MHz
MAC: f0:f5:bd:xx:xx:xx
Uploading stub...
Running stub...
Stub running...
MAC: f0:f5:bd:fc:68:2c
Hard resetting via RTS pin...
```

**Reply by vtraveller** [reply]
Everything seems to be working fine.  I'm still trying to get the power measurement part working, but its not essential.

I do however get this odd message when attempting an OTA update:

`E (582296) boot_comm: Image requires efuse blk rev >= v285.24, but chip is v1.3`

Not sure how to tackle this.

**Reply by MalSmalley** [reply]
Good Work!

The config from mine attached if this helps.   
Also I run it under Node Red with a UIBuilder module giving power and a plot of last hour.
If found that the V & I Limits don't work under Tasmota - they just give an alert but the Power does.
Not tried an OTA update - it works as is so leave it be...

![Screenshot 2025-02-02 at 16 21 45](https://github.com/user-attachments/assets/13efe0eb-5de7-47e5-a09f-6c333bbfbb56)
![Screenshot 2025-02-02 at 16 17 15](https://github.com/user-attachments/assets/c2f1e871-a290-4cd6-af4c-a4c0e7d6bb6a)

**Reply by ChrisVonFrosta** [reply]
your gui implant looks very nice. Would you share your berry file(s)?

**Reply by vtraveller** [reply]
I've just modded a second S60 and it worked flawlessly.  The steps pretty straight forward, but the jig design may need adapting depending on what you have available.

**Parts I used**:
- 16mm long, 0.7mm diameter sprung pogo pins (x3)
- GodHand 1.1mm and 0.7mm drill bits with a pin-wheel
- UK single plug socket

**Pre-Steps** (do these once)
1. Print the two jigs in PLA, reduce the height of the key on the 0.7mm jig if you have shorter pogos like mine
2. Drill out the 1.4mm jig holes with the 1.1mm bit all the way through, smooth the holes to be useable
3. Partially drill out the 0.7mm jig with the 0.7mm bit, such that you can affix the pogos
4. Loop and solder wires around such pogos at the top of the jig
5. Connect up your wires, I used the UK socket to deal with the ground as it was way more stable.  You fit the jig then plug it in

**Steps**
1. Put the 1.4mm jig over the earth, get the 1.1mm bit and gently put shallow holes / divots on the S60.  This will allow the 0.7mm bit to drop into the pre-hole.
2. Take off the jig, then using the 0.7mm bit, continue to drill the three holes as straight as you can. With the pin-wheel drill that's much easier than you think.
3. Take you pogo jig, put it on the earth pin, plug everything in - and you're good to go

Actually steps take barely one minute to complete.  Flashing could take longer. 🤣 

When I have some time, I'll update the printables.com page with photos and clearer steps.

**Reply by vtraveller** [reply]
And finally, this was my design goal. Minimal intrusion and always available if bricked or OTA is acting up. 

![image](https://github.com/user-attachments/assets/21904068-1b2f-4c2f-b2e8-334d5142cedb)

**Reply by vtraveller** [reply]
> Good Work!
> 
> 
> 
> The config from mine attached if this helps.   
> 
> Also I run it under Node Red with a UIBuilder module giving power and a plot of last hour.
> 
> If found that the V & I Limits don't work under Tasmota - they just give an alert but the Power does.
> 
> Not tried an OTA update - it works as is so leave it be...
> 
> 

Much appreciated.

I've custom f/w I've been running on a mix of Sonoff Basic and R2s. Over time their wifi radios just seem to go bad. The rest of it works.

So these S60s are replacing the older failing Basic/R2 ones. I'm quite pleased with the general build of this latest S60 generation, however only time will really answer the reliability question.

If there's any comments about wanting more details on what I did, I'll write more up, but I think the problem is now well and truly solved for me.

**Reply by MalSmalley** [reply]
Very neat job.   I've just ordered 2 more units.  I'm tempted to give @JBFUK 's 'vice' method a go.
Don't possess a 3D printer......

**Reply by vtraveller** [reply]
> Very neat job.   I've just ordered 2 more units.  I'm tempted to give @JBFUK 's 'vice' method a go.
> 
> Don't possess a 3D printer......
> 

Oh the torment. I use my 3D specifically for problem solving. I'm a lousy carpenter and a 3D printer gives me that Swiss-watch precision I craved.

You could technically online print or join a maker club. I know people that do that to work out if 3D printing is for them.

**Reply by vtraveller** [reply]
@MalSmalley DM as vtraveller on discord. I may be able to help.

**Reply by MalSmalley** [reply]
Thanks.  My two new ones popped open no problem & all flashed & working.
A bit cautions with the first one until you calibrate how hard to squeeze the vice but after that its pretty plain sailing.

@vtraveller did you solve your OTA update issue?

**Reply by vtraveller** [reply]
Not yet. I'll look at the weekend. I need to change from SSDP client/server code over to a new UPnP version as discovery is acting up. 

But it's on my list. I think an efuse needs flipping. I'm just hoping they didn't do a perm change otherwise I'm walking around to reflash every time I mod with a jig.

**Reply by vtraveller** [reply]
ooooh this gets well dicey.  Basically I'm trying to track down a newer second stage boot loader (i.e. flash programming mode).

https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/startup.html#second-stage-bootloader

You can download the factory flashing tool [here](https://www.espressif.com/en/support/download/other-tools).

If you want a reading arc:

https://docs.espressif.com/projects/esp-idf/en/v5.0/esp32c3/api-reference/system/efuse.html?form=MG0AV3
https://docs.espressif.com/projects/esp-idf/en/stable/esp32c3/api-reference/system/ota.html?form=MG0AV3
https://docs.espressif.com/projects/esp-test-tools/en/latest/esp32/production_stage/tools/flash_download_tool.html
https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/bootloader.html

If you have more smarts than me, and know where this is, or whether that's the part of the problem I'm trying to solve, let me know.  Either the secondary boot loader needs updating at address 0x1000, or the OTA partition table needs configuring at 0x8000.

There's interesting reading for eFuse rollback at the end of this page.  I wondering if its caught by this and compatibility within the table:

https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/system/ota.html

I think my best bet is to run the otatool.py and see what it says.  Fortunately the entire ESP-IDF is open source, so I could also reverse engineer the error code back to something sane.

https://github.com/espressif/esp-idf

So that's where I am at the moment.  Don't you long for the days of a one button fix?

**Reply by JBFUK** [reply]
@vtraveller I feel like someone has a little too much time on their hands 😄 

Glad my opening method has worked for others too.  I find that approach works for most of these smart plugs.

**Reply by MelanieT** [reply]
I had to move the holes by about 0.6mm, testing with the pogo jig in place and an open unit. I found the best way to "dial in" the pogo jig is to use the drill jig and drill the holes, then place the pogo jig and, with the unit open still, heat the back end of each pogo pin with a soldering iron and move them to aim them precisely, testing by pushing the board down on the pogo and checking they hit the pads. I then fixed the pogo pins in place by encasing the back end, with wires attached, in hot glue.
![IMG_2582](https://github.com/user-attachments/assets/b5ee5233-16ed-41f8-bfa9-3b147eba8bd4)
![IMG_2583](https://github.com/user-attachments/assets/0c6d2e2b-091f-4d1d-ab46-87e6817c5173)
![IMG_2584](https://github.com/user-attachments/assets/219e77af-ed4c-4c00-ab44-19311724dae6)
![IMG_2585](https://github.com/user-attachments/assets/9067a6da-04f6-4b27-9b11-067f9ce6f16e)
Flash successful with the template and firmware above. I have 5 more units on the way now, getting them while the getting is good. These will allow me to match my appliance use to my solar battery capacity.

**Reply by vtraveller** [reply]
Interesting.  Glad to see someone else having a go.

Maybe the size of the pogos, or the printer has some effect.  I did a reprint for the printables.com video and that worked too first go.

I've done 8 so far without issue.  But its good to know your tips in case I hit problems at any point in the future,

Well done.

**Reply by MelanieT** [reply]
I printed mine on a Babycube because it's so small, using 0.4mm nozzle and PETG. The messy looking hole is from correcting the 0.6mm after the fact, after drilling using the original template. Then I imported the STL into Solidworks and rebuilt it to be able to adjust the arc distance of the holes. I removed your "nudge" because my printer deposits enough material to close the hole even with the exact pad location used. I lowered the area with the pogo pins because mine were shorter. I'll get the rest of the plugs tomorrow, if all goes well, i'll post the STL and STEP files.

**Reply by vtraveller** [reply]
> I lowered the area with the pogo pins because mine were shorter. I'll get the rest of the plugs tomorrow, if all goes well, i'll post the STL and STEP files.

Understandable.  You have to work with what you've got.  I made things like that to try and keep it parallel and solid.

Also as my pogos were so long, I could bend them easily for hunting my targets.

That said, I think you're probably learning more from the process than I did.

**Reply by MelanieT** [reply]
If you print a drill jig, don't use it for drilling. 1st drilling worked perfect, 2nd somewhat, by the 3rd, the holes were all over the place and I had to make extra space because the drill had skewed the 3d printed jigs. Either print one jig per plug, or use the jigs for marking only, but then, the angles may be wrong..... In the end, 5 more plugs are tasmotised.

**Reply by vtraveller** [reply]
If you saw my video, I only marked. As I have a pinwheel drill, perpendicular angles are easy to do.

I'm sure the jig design could be improved. Ultimately I was experimenting. The jig needs to be held tight against the plug for marking.

One thing I had considered before what I did, was a larger hole covering all three points. Then designing a connector that mounted for a standard fitting.

Essentially test points flush with the outside. And something easier with less specialised equipment. But my work with my 8 switches was so consistent, I never bothered.

I flash them regularly as OTA seems disabled. I use my own firmware running a serverless mesh.

Never had issues except with the first one. I filled my failed hole with superglue and plastic dust on top then cut it flush to make the restoration clean.

**Reply by vtraveller** [reply]
> The messy looking hole is from correcting the 0.6mm after the fact, after drilling using the original template. 

ok - so I bought another five S60s and started modding them.  The first one worked fine, but the four others failed.  four were without power until I coerced the power pin.  Two had Rx but not Tx.

I put the same jig into the other 9 I have and they worked fine.  So this makes me think there's a twisting tolerance in how they're manufactured.

You never said if the 0.6mm was a rotational shift, but I'm thinking it probably is.  At this point I plan to use a pogo on a multimeter to ground to find where the pads have moved to.

**Reply by vtraveller** [reply]
Fixed.  I enlarged the top-most hole from 0.7mm to 1.1mm.  I then bent the pin slightly towards the centre.  This fixed the majority of problems.  I did bend the middle pin slightly down.

The top-pin is host Tx, and the middle host Rx.  If you have power then you can ignore the bottom-pin as its lined up fine.

If Rx is correct then any serial monitor will show a boot sequence.  If the Tx pin is out then `py esptool.py --port COM6 read_mac` will fail.

I would agree that there is a minor error in tolerance.  I found all existing S60 plugs worked fine on a realigned set of pins.  0.4 to 0.5mm seems to be the margin, so 0.6mm fits with an accurate landing point.

**Comment by MalSmalley** [comment]
@ChrisVonFrosta 
It's html/css/js using UIBuilder component within NodeRed.
It's integrated with Light Switches, various environmental sensors & some other controls I have.

**Comment by divadiow** [comment]
If anyone's got their S60TPG in pieces still would they mind measuring the resistance of these two please by GPIO10?

![image](https://github.com/user-attachments/assets/577ac810-29de-4d3c-aa34-8e32702ade43)

thank you!

**Comment by divadiow** [comment]
offending resistors located. 12k it seems 
![IMG00005](https://github.com/user-attachments/assets/1e786cbd-e86c-4f8b-b47a-6f38e84febbc)
damn these are tiny

**Reply by MelanieT** [reply]
The one on the solder wick looks more like a capacitor to me. They're 0201 parts.

**Comment by aj-way** [comment]
@MalSmalley great guide thank you.

**Comment by amirmazmi** [comment]
Thank you for this discussion. Helped with a few gotchas.   
  
Managed to flash 2 units. The hardest part for me was prying them open but clamping and rotating works. i cheated a bit and used a spudger (clamp is not strong enough). Helps it pop open while being clamped.  
     
Shell is made out of polycarbonate, do not use isopropyl alcohol or acetone as it will make it brittle and crack. 

To close it, I applied a thin slither of super glue around the rim. The idea is that I would be able to break the glue easily as super glue is brittle. If it drops and becomes loose, could just as easily reapply. 
  
  
<img width="400" height="1122" alt="test_probe" src="https://github.com/user-attachments/assets/66958e36-5771-4b44-ac10-8aaf2bb045b2" />


<img width="1462" height="822" alt="flash_setup" src="https://github.com/user-attachments/assets/af17587d-ead8-4af5-aebe-c249fc463043" />

**Comment by shoog03** [comment]
Thanks for the guidance I found it very useful.
A few observations from  my own experience
-when cracking the lid a vice works great but you need to squeeze the pin base till it deforms very slightly before it will crack the glue.
-once the lid is off I simply leavered the circuit board off its two white pins and upwards. It then clicked into place on the lip of the plastic earth pin shroud and held there. This gave me enough room to solder very fine wires onto the RX TX and R15 pins using a fine tipped soldering iron. 

I managed to flash the plug in around half an hour from start to finish and with a bit of superglue you wouldn't know that the case had been opened up. Not the easiest - but certainly not the hardest.

**Comment by HG0161** [comment]
There's a guide on the my git for OTA [HERE](https://github.com/HG0161/SonoffS60OTA) It has a lot of potential to be streamlined but for now it remains as running a bunch of scripts and some router configuration.

The next thing i want to do is re-partition the two current partitions so we can reclaim more space for the full tasmota32c3.bin as it's currently 2,180,992 bytes, 149,376 bytes too large. If anyone can obtain the actual map would be much appreciated

**Reply by amirmazmi** [reply]
OTA would be really nice but at least for me personally I am trying to stay clear of the ewelink app or connecting the device to the internet

**Comment by npa-homeassistant** [comment]
I'm trying to prise one of these open and not getting very far. Which face is glued and where do I need to be compressing it, please? 

TIA

**Reply by npa-homeassistant** [reply]
Ahh.. Belay that. Just as I pressed Send, it popped.

**Reply by npa-homeassistant** [reply]
...ouch and then cracked.

**Comment by sfromis** [comment]
Maybe consider the OTA flashing method HG0161 is working on. I'd guess it to work for the same for the UK variant too, but I've not tried it myself. https://github.com/arendst/Tasmota/discussions/21255#discussioncomment-18239154
